### PR TITLE
Eddie

### PR DESCRIPTION
**Pull request by EddieFerro**

### CODE REVIEW

File: character.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+#enemy class can be later subclassed to create specific types of enemies, where some values might be more static
+class Enemy(Character):
+    def __init__(self, health, fire_rate, spawnx, spawny, velx, vely):
+        super().__init__(health)
+        self._fire_rate = fire_rate
+        self._velx = velx
+        self._vely = vely
+        self._spawnx = spawnx
+        self._spawny = spawny
```
Comment (KaytonFletcher) [inline character.py:83]: Honestly not sure these need to be class variables. Just used to set initial sprite position. Could just be passed to sprite creation function

File: character.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+
+class Player(Character):
+    def __init__(self, health, fire_rate, spawnx, spawny, velx, vely):
+        super().__init__(health)
+        self.fire_rate = fire_rate
+        self.velx = velx
+        self.vely = vely
+        self._spawnx = spawnx
+        self._spawny = spawny
```
Comment (KaytonFletcher) [inline character.py:26]: See comment below on enemy spawn. Same thing

File: character.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        return self._health
+    
+    @health.setter
+    def health(self, health):
+        self._health = health
+
+    def _sprite_creation(self):
+        print("Test")
+
+
```
Comment (KaytonFletcher) [inline character.py:18]: In general, Enemy and Player share velocity, sprite creation and fire rate. So these things should be abstracted to the character class

**Review by KaytonFletcher — APPROVED** [review]
I'm going to approve this so we can move forward and add features. But we should definitely fix a few things with this, at least what I left comments on.

File: character.py (hunk 3 above)
Comment (EddieFerro) [reply character.py:18]: the only reason i wasnt sure about putting them together is that enemy fire rate, and velocity might not change but the player will have those values changed so i was not sure how properties relates through inheritance in python

**Pull request merged**